graphite-ch-optimizer.go

```diff
@@ -345,6 +362,7 @@ func optimize() error {
 	logrus.Infof("Merges will be applied: %d", len(merges))
 
 	for _, m := range merges {
+		m := m
```

Comment:
Is it made to preserve the value despite possible changes of `merges`?

Reply:
I was burnt a few times with this language _feature_. Go is reusing `m` in `for` which can cause some unexpected behaviour in case you start using routines or closures. I have written a small [example](https://play.golang.org/p/Xgq7lVehoCI) for you to try. 

This is not causing troubles with the current implementation so feel free to tell me that I am overcautious and I should drop that commit.

Another detailed explanation of this [gotcha](http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/index.html#closure_for_it_vars).

Reply:
Thank you, the example is very self-descriptive!